assets/js/script.js

```diff
@@ -1,51 +1,111 @@
-const locationH2 = document.createElement("h2")
-weatherFeaturesEl.appendChild(locationH2)
-
-
-
-
-const weatherFeaturesSection = document.createElement("section")    
-weatherFeaturesSection.setAttribute("class", "")
-
-weatherFeaturesSection.setAttribute("id", "weather-features")
-
-const weatherFeaturesArticle = document.createElement("article")
-weatherFeaturesArticle.setAttribute("class", "media px-4")
-
-const weatherFeaturesFigure = document.createElement("figure")
-weatherFeaturesFigure.setAttribute("class", "media-left")
-
-const weatherFeaturesP = document.createElement("p")
-weatherFeaturesP.setAttribute("class", "icon")
-
-i
-
-const weatherFeaturesDiv = document.createElement("div")
-weatherFeaturesDiv.setAttribute("class", "media-content")
-
 
+var dateInputEl = document.querySelector("#date-input");
+var addressInputEl = document.querySelector("#location-input");
+var locationSelectContainer = document.querySelector("#date-and-location-form");
+var weatherFeaturesEl = document.querySelector("#weather-features");
+var whenAndWhereContainerEl = document.querySelector("#when-and-where");
+
+var locationSubmitHandler = function (event) {
+  // prevent page from refreshing
+  event.preventDefault();
+
+  // get value from input element
+  let address = addressInputEl.value.trim();
+  let date = dateInputEl.value.trim();
+
+  if (date && address) {
+    getWeatherData(date, address);
+
+    // clear old content
+    weatherFeaturesEl.textContent = "";
+    whenAndWhereContainerEl.textContent = "";
+  } else {
+    alert("Please enter a valid date");
+  }
+};
+
+var getWeatherData = function (date, address) {
+  const APIKEY = "TG354CRJY4Z63JNSXQG9GRMXE";
+  // format the Visual Crossing API url - returned data in US units
+  var apiUrl = `https://weather.visualcrossing.com/VisualCrossingWebServices/rest/services/weatherdata/history?&aggregateHours=24&startDateTime=${date}T00:00:00&endDateTime=${date}T00:00:00&unitGroup=us&contentType=json&dayStartTime=0:0:00&dayEndTime=0:0:00&location=${address}&key=${APIKEY}`; // update address to use Mapbox API
+
+  // make a fetch request to url
+  fetch(apiUrl)
+    .then(function (response) {
+      // request was successful
+      if (response.ok) {
+        return response.json()
+      } else {
+        throw new Error("Error: " + response.statusText)
+      }
+    })
+    .then(function (fetchData) {
+      displayWeatherData(fetchData, date, address);
+    })
+    .catch(function (error) {
+      alert("Unable to connect to Visual Crossing");
+    });
+};
+
+var displayWeatherData = function (fetchData, date, address) {
+  // check if api returned any repos
+  if (!fetchData) {
+    whenAndWhereContainerEl.textContent = "No weather data found. Update your search parameters and please try again.";
+    return;
+  }
+
+  console.log(fetchData);
+  whenAndWhereContainerEl.innerHTML = `${date} in ${fetchData.locations[address].name}`;
```

Comment:
🙌 Weather feature element creation and display works great, looks great

`whenAndWhereContainerEl.innerHTML` is being set with unstyled text on line 58.  Do we want to replace it with this from your earlier commit?

```
const locationH2 = document.createElement("h2");
locationH2.setAttribute("class", "is-size-4");
locationH2.textContent = `${date} in ${fetchData.locations[address].name}`;
whenAndWhereEl.appendChild(locationH2);
```

Reply:
+1 I did that more as a placeholder initially so feel free to remove.